### PR TITLE
feat: add callback function for version history component mounting

### PR DESCRIPTION
**Pull request by jlarsen-box**

### CODE REVIEW

File: src/elements/content-sidebar/versions/flowTypes.js [hunk 1]
```diff
@@ -4,3 +4,4 @@ import type { BoxItemVersion } from '../../../common/types/core';
 
 export type VersionActionCallback = (versionId: string) => void;
 export type VersionChangeCallback = (version: ?BoxItemVersion, additionalVersionInfo: ?AdditionalVersionInfo) => void;
+export type VersionHistoryLoadCallback = (data: ?Object) => void;
```
Comment (jackiejou) [inline src/elements/content-sidebar/versions/flowTypes.js:7]: should data have a type? UpsellType?

File: src/elements/content-sidebar/versions/VersionsSidebarContainer.js [hunk 2]
```diff
@@ -82,8 +84,10 @@ class VersionsSidebarContainer extends React.Component<Props, State> {
     window: any = window;
 
     componentDidMount() {
+        const { versionUpsellExperience, onLoad } = this.props;
         this.initialize();
         this.fetchData();
+        onLoad('preview', 'versions', { versionUpsellExperience });
```
Comment (jstoffan) [inline src/elements/content-sidebar/versions/VersionsSidebarContainer.js:90]: Should we wait until the data has loaded? If so, let's update `fetchData` to return its promise and then do this:

```suggestion
        this.fetchData()
            .then(() => {
                onLoad('preview', 'versions', { versionUpsellExperience });
            });
```

File: src/elements/content-sidebar/versions/__tests__/VersionsSidebarContainer.test.js [hunk 3]
```diff
@@ -32,6 +32,7 @@ describe('elements/content-sidebar/versions/VersionsSidebarContainer', () => {
     const api = {
         getVersionsAPI: () => versionsAPI,
     };
+
```
Comment (jstoffan) [inline src/elements/content-sidebar/versions/__tests__/VersionsSidebarContainer.test.js:35]: Revert? Do we have any tests for the `onLoad` call?

File: src/elements/content-sidebar/versions/flowTypes.js [hunk 4]
```diff
@@ -4,3 +4,4 @@ import type { BoxItemVersion } from '../../../common/types/core';
 
 export type VersionActionCallback = (versionId: string) => void;
 export type VersionChangeCallback = (version: ?BoxItemVersion, additionalVersionInfo: ?AdditionalVersionInfo) => void;
+export type onLoadCallback = (component: string, feature: string, data?: Object) => void;
```
Comment (jstoffan) [inline src/elements/content-sidebar/versions/flowTypes.js:7]: How about `SidebarLoadCallback` to be consistent with the types above?

**Review by jstoffan — COMMENTED** [review]
Make sure to update the PR title and commit messages to conform to the commit message requirements.

File: src/elements/content-sidebar/versions/flowTypes.js (hunk 4 above)
Comment (jlarsen-box) [reply src/elements/content-sidebar/versions/flowTypes.js:7]: Ack!

File: src/elements/content-sidebar/versions/__tests__/VersionsSidebarContainer.test.js (hunk 3 above)
Comment (jlarsen-box) [reply src/elements/content-sidebar/versions/__tests__/VersionsSidebarContainer.test.js:35]: I will likely add one to to the code in EUA? unless you think its a good idea to have one here too.

File: src/elements/content-sidebar/versions/VersionsSidebarContainer.js (hunk 2 above)
Comment (jlarsen-box) [reply src/elements/content-sidebar/versions/VersionsSidebarContainer.js:90]: ah, good idea. Will update

File: src/elements/content-sidebar/versions/__tests__/VersionsSidebarContainer.test.js (hunk 3 above)
Comment (jstoffan) [reply src/elements/content-sidebar/versions/__tests__/VersionsSidebarContainer.test.js:35]: We generally want to have unit tests on both sides to cover the logic for each individual component.

Comment (jlarsen-box) [reply src/elements/content-sidebar/versions/__tests__/VersionsSidebarContainer.test.js:35]: ack, will add one here too :)

File: src/elements/content-sidebar/versions/VersionsSidebarContainer.js [hunk 5]
```diff
@@ -82,8 +84,11 @@ class VersionsSidebarContainer extends React.Component<Props, State> {
     window: any = window;
 
     componentDidMount() {
+        const { onLoad, versionUpsellExperience } = this.props;
         this.initialize();
-        this.fetchData();
+        this.fetchData().then(() => {
+            onLoad('preview', 'versions', { versionUpsellExperience });
```
Comment (jstoffan) [inline src/elements/content-sidebar/versions/VersionsSidebarContainer.js:90]: How about something like, `onLoad({ component: 'preview', feature: 'versions', versionUpsellExperience })`

**Review by jstoffan — COMMENTED** [review]
LGTM, just need a test.

File: src/elements/content-sidebar/versions/VersionsSidebarContainer.js (hunk 5 above)
Comment (jlarsen-box) [reply src/elements/content-sidebar/versions/VersionsSidebarContainer.js:90]: right now onLoad takes two strings and an object. Do you think we should have it only take an object, and put the component and feature strings in the data object?

**Review by jstoffan — APPROVED** [review]

**Review by sanjui9000 — APPROVED** [review]

**Pull request merged**